packages/block-library/src/query/edit/pattern-selection.js

```diff
@@ -43,7 +43,16 @@ export function useBlockPatterns( clientId, attributes ) {
 		clientId,
 		attributes
 	);
-	return usePatterns( clientId, blockNameForPatterns );
+	const allPatterns = usePatterns( clientId, blockNameForPatterns );
+	// Filter out any patterns that don't have Query as their root block
+	// so that a Query block is always replaced by another Query block.
+	const rootBlockPatterns = useMemo( () => {
+		return allPatterns.filter( ( pattern ) => {
+			return pattern.blocks?.[ 0 ]?.name === blockNameForPatterns;
+		} );
+	}, [ allPatterns, blockNameForPatterns ] );
```

Comment:
This is working pretty well for me, and I think the trade off (that it _might_ reduce the number of available patterns) is probably worth it. 

Do you think there's a case for best practice eduction that Query patterns should ideally be wrapped in a Query block?

Reply:
I don't know. Conceivably you may want to create a complex pattern with a query in it, or even multiple queries. I think there's a place for those, but they aren't viable as replacements for a single Query block.